### ISSUE TITLE
"Save as" from the svg view creates invalid svgs

### ISSUE THREAD

**Issue by uwe83** [hallvard/plantuml#177]
**1. Create text file in Eclipse containing almost any diagram. E.g. the classic test.txt containing:**

Case "trivial"
```
@startuml
alice -> bob : hello
@enduml
```
or, in Case "complex"
```
@startuml

actor Utilisateur as user
participant "formSign.js" as form <<Contrôleur formulaire>>
participant "Sign.java" as controler <<(C,#ADD1B2) Contrôleur formulaire>>
participant "Secure.java" as secure <<(C,#ADD1B2) authentification>>
participant "Security.java" as security <<(C,#ADD1B2) sécurité>>

box "Application Web" #LightBlue
	participant form
end box

box "Serveur Play" #LightGreen
	participant controler
	participant secure
	participant security
end box

user -> form : submitSignIn()
form -> form : getParameters()
form -> form : result = checkFields()

alt result

    form -> controler : formSignIn(email,pwd)
    controler -> controler : result = checkFields()
    
    alt result
    	controler -> secure : Secure.authenticate(email, pwd, true);
    	secure -> security : onAuthenticated()
    	security --> form : renderJSON(0);
    	form --> user : display main page
    else !result
    	controler --> form : renderJSON(1)
    	form --> user : display error
    end
    
    Utilisateur -> Utilisateur : Bla 
    
else !result
	form --> user : display error
end

@enduml
```

**2. Right click into the picture and select "Save as"**

![image](https://github.com/hallvard/plantuml/assets/159607470/0ad63a38-76b9-450b-9886-14b4fd4b3edd)

**3a. save the file into e.g. "trivial.svg"**
![trivial](https://github.com/hallvard/plantuml/assets/159607470/831df41c-bd21-41d9-ba8b-08bfde374975)

**3b save the file into e.g. "complex.svg"**
![complex](https://github.com/hallvard/plantuml/assets/159607470/4c79352b-d166-46d9-9920-6d0926bfd1b0)

**4. The resulting svg is invalid according to Edge and Chrome:**
in both cases

**Editing the file shows it contains a duplicate </defs> closing tag in column 603 as indicated by the browsers:**

![image](https://github.com/hallvard/plantuml/assets/159607470/52afe6fe-b46c-4f76-bed9-62e9f5d8f055)

**Removing the "extra"</defs> tag using a text editor results in**
Case "trivial": the browser no longer complaining, but rendering "nothing", everything white
Case "complex": the browser rendering the image OK

**Comment by travkin79** [comment]
Hi @uwe83,
I'm working on a new release with newer PlantUML lib version 1.2024.4. It seems, I cannot reproduce this issue with the latest version from master branch. Could you please test the PlantUML Eclipse plug-ins from current master branch? Thank you.

In my version, I don't have a save file entry in the context menu in the PlantUML SVG view. Instead, I used the export feature from the context menu in the PlantUML view and saved the diagram as *.svg file.

**Comment by uwe83** [comment]
Hello Dietrich,

thank you for looking into this. From my understanding I would need to
build the plugin myself using Maven. I need to learn how to do that. I
tried this morning but could not get there yet. I will retry later. I
found /net.sourceforge.plantuml.releng/net.sourceforge.plantuml.composite/README.md
and assume this is what I need to follow. I also found
https://github.com/hallvard/plantuml/issues/122 and will look at that.

Am Mi., 10. Apr. 2024 um 17:58 Uhr schrieb Dietrich Travkin <
***@***.***>:

**Comment by travkin79** [comment]
Hi @uwe83,
The file you found `net.sourceforge.plantuml.releng/net.sourceforge.plantuml.composite/README.md` is the right source, the issue #122 seems too old.

But I could also provide an already built version to you. I built it anyway for local testing.

**Comment by travkin79** [comment]
I've attached an update site with only the latest release candidate. Could you unzip and test it? You'll need to add the update site with a URL of the form `file:/path/to/your/local/update-site-directory`
[update site.zip](https://github.com/hallvard/plantuml/files/14944413/update.site.zip)

**Comment by uwe83** [comment]
I installed it successfuly, but I cannot find any evidence for it except for the installation details:
![image](https://github.com/hallvard/plantuml/assets/159607470/6e0d886a-5000-4729-a3b1-cd4e6c1ebe29)
I find no PlantUML preferences and no PlantUML views :-(.

I first installed this into my existing environment. After the restart the Plantuml views were still open, but had seized working. I then installed once more into a freshly installed Eclipse https://www.eclipse.org/downloads/download.php?file=/technology/epp/downloads/release/2024-03/R/eclipse-dsl-2024-03-R-win32-x86_64.zip. 

If there's anything I can do to help find out what is wrong here, please let me know.

**Comment by travkin79** [comment]
Hi @uwe83 ,
Could you please take a look into the Error Log View and check if there are any new entries. If you see any PlantUML-related entries, please let me know and add the exception stacktrace if available.

**Comment by travkin79** [comment]
Hello @uwe83,
I installed another Eclipse IDE on a Windows PC and installed PlantUML Eclipse Plug-ins from my `update site.zip` file and the views are still there and work fine.

But the issue you described, remains. What I observed is that it only occurs on Windows. In fact, in PlantUML SVG view does only offer the "Save as..." context menu entry on Windows, not on MacOS and not on Linux. Besides, if you instead use the export feature from the PlantUML view (not SVG in view name) and save the file as *.svg, then the described problem does not appear, there is only one occurrence of the `<defs/>` tag.

**Comment by uwe83** [comment]
Yes, that is in fact weird. But it's good to know the workaround via the PlantUML view! Thanks!
